pkg/kubectl/cmd/proxy.go

```diff
@@ -83,14 +83,33 @@ func NewCmdProxy(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringP("address", "", "127.0.0.1", "The IP address on which to serve on.")
 	cmd.Flags().Bool("disable-filter", false, "If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks, when used with an accessible port.")
 	cmd.Flags().StringP("unix-socket", "u", "", "Unix socket on which to run the proxy.")
+	cmd.Flags().StringP("curl", "", "", "Raw URI to request from the proxy server.")
 	return cmd
 }
 
+func curl(f *cmdutil.Factory, uri, address string, port int) error {
+	restClient, err := f.RESTClientProxy(address, port)
+	if err != nil {
+		return err
+	}
+
+	return cmdutil.PrintRESTClientStream(restClient, uri)
+}
+
 func RunProxy(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
 	path := cmdutil.GetFlagString(cmd, "unix-socket")
 	port := cmdutil.GetFlagInt(cmd, "port")
 	address := cmdutil.GetFlagString(cmd, "address")
 
+	uri := cmdutil.GetFlagString(cmd, "curl")
+
+	if uri != "" {
+		if path != "" {
+			return errors.New("Don't specify both --curl and --unix-socket, --curl doesn't work with unix sockets.")
```

Comment:
Can you spawn an issue?  Because I would rather fix that than add an error here.  Unix sockets are EXACTLY what we want to be able to do for local hosts - the security risks of kube proxy are immense in general practice.